tests/invalid_models_tests/test_models.py

```diff
@@ -1823,3 +1974,31 @@ class Meta:
                 ]
 
         self.assertEqual(Model.check(databases=self.databases), [])
+
+    @unittest.skipIf(
+        connection.vendor == 'mysql',
+        'MySQL does not support unique constraints with conditions',
+    )
+    def test_unique_constraint_condition_with_join(self):
+        class Model(models.Model):
+            age = models.PositiveSmallIntegerField()
+            parent = models.ForeignKey('self', models.CASCADE)
+
+            class Meta:
+                required_db_features = {'supports_partial_indexes'}
+                constraints = [
+                    models.UniqueConstraint(
+                        name='name',
+                        fields=['age'],
+                        condition=models.Q(parent__age__lt=2),
+                     ),
+                 ]
+
```

Comment:
Please fix indentation in these lines (see flake failures).

Reply:
Sorry for that, I was checking the wrong file :man_facepalming: